my_enumerables.rb

```diff
@@ -0,0 +1,185 @@
+# frozen_string_literal: true
+
+# rubocop : disable Metrics/ModuleLength
+# rubocop : disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop : disable Metrics/MethodLength
+# rubocop : disable Style/Documentation
+# rubocop : disable Metrics/AbcSize
+
+module Enumerable
+  def my_each
+    if is_a?(Array)
+      0.upto size - 1 do |val|
+        yield(self[val]) if block_given?
+      end
+    elsif is_a?(Range)
+      each do |val|
+        yield(val) if block_given?
+      end
+    else
+      hash_keys = keys
+      0.upto hash_keys.length - 1 do |num|
+        key = hash_keys[num]
+        yield(key, self[key])
+      end
+    end
+  end
+
+  def my_each_with_index
+    0.upto size - 1 do |val|
+      yield(self[val], val)
+    end
+  end
+
+  def my_select
+    return_arr = []
+    return_hash = {}
+    if is_a?(Hash)
+      my_each do |key, val|
+        return_hash[key] = val if yield(key, val)
+      end
+      return_hash
+    else
+      my_each do |val|
+        return_arr.push(val) if yield(val)
+      end
+      return_arr
+    end
+  end
```

Comment:
* [ ] should have working `my_select` identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_select returns an enumerator if no block is given
  ```
    array = [1,2,5,8,7,98,4]
    p array.my_select.class # not Enumerator
  ```